### PR TITLE
Additional links json file

### PR DESCRIPTION
**Pull request by Vedant1202**
JSON file for additional links

## Summary

JSON file for configuring additional links as described in [caMicroscope#468](https://github.com/camicroscope/caMicroscope/issues/468). 

The links are customizable with the following parameters:
1. The URL
2. The name to be displayed
3. The icon to be displayed (material icons)
4. A boolean to choose if the link is to be opened in new tab, or in current tab.

This feature requires [caMicroscope PR-511](https://github.com/camicroscope/caMicroscope/pull/511) to work.

## Motivation

Fixes [caMicroscope#468](https://github.com/camicroscope/caMicroscope/issues/468). 

## Testing

This has been tested on local system running Ubuntu 18.04 and docker-compose version 1.28.6

## Questions

Yes, I am unsure if I should add the relevant documentation describing the structure of the JSON data in the README. Or should I create a new README file for that.

### CODE REVIEW

**Comment by Vedant1202** [comment]
@birm This needs to be merged as well. Also do you want me to add some description for the same in the README?

**Comment by birm** [comment]
Moreso, the back service needs a mounted volume for the config, such as: 
```yml
    volumes:
      - ./config/login.html:/src/static/login.html
      - ./jwt_keys/:/src/keys/
      - ./config/routes.json:/src/routes.json
      - ./config/contentSecurityPolicy.json:/src/contentSecurityPolicy.json
      - ./config/additional_links.json:/src/static/additional_links.json
   ```

**Comment by Vedant1202** [comment]
Ahh right right, I forgot to add those to the commit

**Comment by Vedant1202** [comment]
@birm I updated it. Should I update to the README ?

File: config/additional_links.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        "displayName":"Sample link 2",
+        "url":"http://sample-link-2.com",
+        "icon": "speaker_notes"
+    },
+    {
+        "displayName":"Sample link 3",
+        "url":"http://sample-link-3.com",
+        "openInNewTab": false
+    }
+]
```
Comment (birm) [inline config/additional_links.json:17]: following https://github.com/camicroscope/caMicroscope/commit/9291074f2d274dcded4e9938800c04df7e0cc379

```suggestion
[
    {
        "displayName":"Bug Report",
        "url":"https://goo.gl/forms/mgyhx4ADH0UuEQJ53",
        "icon": "bug_report"
    }
]
```

File: develop.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ services:
       - mongo
   back:
     build:
-      context: "https://github.com/camicroscope/caracal.git#develop"
+     context: "https://github.com/camicroscope/caracal.git#v3.9.1"
```
Comment (birm) [inline develop.yml:21]:
```suggestion
     context: "https://github.com/camicroscope/caracal.git#develop"
```

File: config/additional_links.json (hunk 1 above)
Comment (Vedant1202) [reply config/additional_links.json:17]: @birm Shouldn't it contain an example case where `openInNewTab` is set to `false`, for clarity?

Comment (birm) [reply config/additional_links.json:17]: we could set it to true, which should still show off the field

Comment (Vedant1202) [reply config/additional_links.json:17]: Alright, I'll do that

**Comment by Vedant1202** [comment]
@birm Done

File: develop.yml [hunk 3]
```diff
@@ -18,7 +18,7 @@ services:
       - mongo
   back:
     build:
-      context: "https://github.com/camicroscope/caracal.git#develop"
+     context: "https://github.com/camicroscope/caracal.git#develop"
```
Comment (birm) [inline develop.yml:21]: compose has complained about this missing space in the yml
```suggestion
      context: "https://github.com/camicroscope/caracal.git#develop"
```

**Pull request merged**